air-quality-ui/system_tests/ui/city_page/city_page.spec.ts

```diff
@@ -519,3 +522,180 @@ test.describe('Charts are fully visible in 1920x1080 viewport', () => {
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 25,
+        pm10: 50,
+        so2: 350,
+        no2: 120,
+        o3: 130,
+        site_name: 'AQI 3 site',
+      }),
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 50,
+        pm10: 100,
+        so2: 500,
+        no2: 230,
+        o3: 240,
+        site_name: 'AQI 4 site',
+      }),
+
+      //AQI 2 values
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T01:00:00Z',
+        pm2_5: 20,
+        pm10: 40,
+        so2: 200,
+        no2: 90,
+        o3: 100,
+        site_name: 'Tijuca',
+      }),
+      //AQI 3 values
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T02:00:00Z',
+        pm2_5: 25,
+        pm10: 50,
+        so2: 350,
+        no2: 120,
+        o3: 130,
+        site_name: 'Tijuca',
+      }),
+      //AQI 4 values
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T03:00:00Z',
+        pm2_5: 50,
+        pm10: 100,
+        so2: 500,
+        no2: 230,
+        o3: 240,
+        site_name: 'Tijuca',
+      }),
+      //AQI 5 values
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T04:00:00Z',
+        pm2_5: 75,
+        pm10: 150,
+        so2: 750,
+        no2: 340,
+        o3: 380,
+        site_name: 'Tijuca',
+      }),
+      //AQI 6 values
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T05:00:00Z',
+        pm2_5: 800,
+        pm10: 1200,
+        so2: 1250,
+        no2: 1000,
+        o3: 800,
+        site_name: 'Tijuca',
+      }),
+    ]
+
+    await setupPageWithMockData(page, [
+      {
+        endpointUrl: '*/**/air-pollutant/forecast*',
+        mockedAPIResponse: mockedForecastResponse,
+      },
+      {
+        endpointUrl: '*/**/air-pollutant/measurements*',
+        mockedAPIResponse: mockedMeasurementsCityPageResponse,
+      },
+    ])
+    await gotoPage(page, '/city/Rio%20de%20Janeiro')
+    await cityPage.waitForAllGraphsToBeVisible()
+    await cityPage.setBaseTime('01/07/2024 00:00')
+  })
+
+  test('Expect in-situ AQI 3 plot at 00:00 due to PM2.5', async ({
+    cityPage,
+  }) => {
+    const chartShot = await cityPage.captureChartScreenshot(cityPage.aqiChart)
+    await expect(chartShot).toMatchSnapshot('Midnight-AQI-3-by-PM2.5.png')
+  })
+
+  test('Expect in-situ AQI 2 when AQI site 4 is removed due to PM2.5', async ({
+    cityPage,
+  }) => {
+    await cityPage.siteRemover('AQI 4 site')
+    const chartShot = await cityPage.captureChartScreenshot(cityPage.aqiChart)
+    await expect(chartShot).toMatchSnapshot('Midnight-AQI-2-by-PM2.5.png')
+  })
+
+  test('Expect in-situ AQI at 00:00 to be 3 when a site is removed and reselected', async ({
+    cityPage,
+    page,
+  }) => {
+    await cityPage.siteRemover('AQI 4 site')
```

Comment:
This may be unorthodox. Essentially there was trouble in reselecting the button through automation. Refreshing the page actually reloads with all sites available again. Happy to spend some more time to figure out the dropdown properly 😎

Reply:
Updated with locators. As reloading the page hack may stop working if setting start to become cached etc